### PR TITLE
fixed errors mentioned in PR comments

### PR DESCRIPTION
**Pull request by mghiremath**

### CODE REVIEW

**Review by sanjeet-jain — CHANGES_REQUESTED** [review]
you forgot to test the modal with the backedn it seems and also forgot to add modal button to the week view

File: data/tasks.js [hunk 1]
```diff
@@ -25,7 +25,6 @@ const tasksDataFunctions = {
    * @param {string} title - The title of the task
    * @param {string} textBody - The body text of the task
    * @param {Date} dateAddedTo - The date when the task was added to the collection
-   * @param {Date} dateDueOn - The date when the task is due
    * @param {number} priority - The priority of the task (1, 2, or 3)
    * @param {string} tag - The custom tag for the task
    */
```
Comment (sanjeet-jain) [inline data/tasks.js:30]: update this as well

File: data/tasks.js [hunk 2]
```diff
@@ -50,20 +49,22 @@ const tasksDataFunctions = {
...
     );
     utils.validateDate(dateAddedTo, "dateAddedTo");
-    utils.validateDate(dateDueOn, "dateDueOn");
     utils.validatePriority(priority);
     utils.validateStringInputWithMaxLength(
       tag,
       "tag",
       constants.stringLimits["tag"]
     );
+    checked = utils.validateBooleanInput(checked, "checked");
```
Comment (sanjeet-jain) [inline data/tasks.js:58]: what if i dont check the checkbox in the UI do we need this validation then ?
add an if statement to check if the value is coming or no and if its coming then do the validation
if its undefined set it to false

File: public/scripts/week-calendar.js [hunk 3]
```diff
@@ -111,8 +111,7 @@ function populateTasksModal(userId, taskId) {
       // it needs an iso string
       event_modal.querySelector("input#task_dateAddedTo").value =
         data.dateAddedTo;
-
-      event_modal.querySelector("input#task_dateDueOn").value = data.dateDueOn;
+      event_modal.querySelector("input#task_checked").checked = data.checked;
     },
     error: function (data) {
       resultDiv = document.getElementById("task-update-result");
```
Comment (sanjeet-jain) [inline public/scripts/week-calendar.js:117]: missing let keyword 
let resultDiv

File: public/scripts/week-calendar.js [hunk 4]
```diff
@@ -359,7 +358,7 @@ function onTaskModalClose() {
       event_modal.querySelector("input#task_tag").value = "";
       event_modal.querySelector("select#task_priority").value = "";
       event_modal.querySelector("input#task_dateAddedTo").value = "";
-      event_modal.querySelector("input#task_dateDueOn").value = "";
+      event_modal.querySelector("input#task_checked").value = "";
       resultDiv = document.getElementById("task-update-result");
```
Comment (sanjeet-jain) [inline public/scripts/week-calendar.js:362]: missing let keyword

Comment (sanjeet-jain) [reply public/scripts/week-calendar.js:362]: on line 350 as well

File: public/scripts/week-calendar.js [hunk 5]
```diff
@@ -545,14 +544,13 @@ function submitTaskForm() {
...
       }
-      //todo validations
       if (checkTaskValidations(event.target)) {
         $.ajax({
           method: reqType,
           url: ajaxURL,
           data: jsonData,
           success: function (data) {
-            resultDiv = document.getElementById("task-update-result");
+            let resultDiv = document.getElementById("task-update-result");
```
Comment (sanjeet-jain) [inline public/scripts/week-calendar.js:553]: missing let keyword on line 530

File: routes/tasks.js [hunk 6]
```diff
@@ -45,21 +44,24 @@ router
...
       utils.validatePriority(priority);
       utils.validateStringInputWithMaxLength(
         tag,
         "tag",
         constants.stringLimits["tag"]
       );
+      if (typeof checked === "undefined") {
+        checked = false;
+      }
+      utils.validateBooleanInput(checked, "checked");
```
Comment (sanjeet-jain) [inline routes/tasks.js:56]: this looks good can be used for the other review comments above

File: views/partials/modals/tasksModal.handlebars [hunk 7]
```diff
@@ -69,19 +69,12 @@
...
-            </div>
-
             {{! add more fields for other events }}
+            <div class="mb-3">
+              <label for="task_checked" class="form-label">Completed</label>
+              <input type="checkbox" class="form-check-input" name="checked" id="task_checked" value="true" />
+              <div id="task_checked_error" class="invalid-feedback">
+              </div>
 
           </form>
```
Comment (sanjeet-jain) [inline views/partials/modals/tasksModal.handlebars:79]: check html page with w3 schools validator once to make sure we dont have any duplicate ids or incorrect html
https://validator.w3.org/nu/#textarea

**Review by sanjeet-jain — CHANGES_REQUESTED** [review]

File: public/scripts/week-calendar.js [hunk 8]
```diff
@@ -562,22 +560,19 @@ function submitTaskForm() {
             setTimeout(location.reload.bind(location), 3000);
```
Comment (sanjeet-jain) [inline public/scripts/week-calendar.js:560]: on the live above shouldnt it be populateTasksModal(data.userId, data._id ) ???

File: views/partials/modals/tasksModal.handlebars (hunk 7 above)
Comment (mghiremath) [reply views/partials/modals/tasksModal.handlebars]: valid source now

File: routes/tasks.js (hunk 6 above)
Comment (mghiremath) [reply routes/tasks.js]: changed

File: public/scripts/week-calendar.js (hunk 5 above)
Comment (mghiremath) [reply public/scripts/week-calendar.js]: done

File: public/scripts/week-calendar.js (hunk 4 above)
Comment (mghiremath) [reply public/scripts/week-calendar.js]: done
done

File: public/scripts/week-calendar.js (hunk 3 above)
Comment (mghiremath) [reply public/scripts/week-calendar.js]: done

File: data/tasks.js [hunk 9]
```diff
@@ -105,22 +106,25 @@ const tasksDataFunctions = {
...
-    updatedTask.textBody = updatedTask.textBody.trim();
-    updatedTask.tag = updatedTask.tag.trim().toLowerCase();
+    updatedTaskData = { ...task };
+    updatedTaskData.expired = false;
+    if (updatedTask.checked) {
+      updatedTaskData.checked = utils.validateBooleanInput(
+        updatedTask.checked,
+        "checked"
+      );
+    }
```
[the remark this answers is not in the document]
Comment (mghiremath) [reply data/tasks.js]: refactored

File: data/tasks.js [hunk 10]
```diff
@@ -166,14 +160,17 @@ const tasksDataFunctions = {
...
     }
 
     if (typeof updatedTask.checked !== "undefined") {
-      utils.validateBooleanInput(updatedTask.checked, "checked");
-      updatedTaskData.checked = updatedTask.checked;
+      updatedTaskData.checked = utils.validateBooleanInput(
+        updatedTask.checked,
+        "checked"
+      );
+      updatedTaskData.expired = updatedTaskData.checked ? true : false;
```
[the remark this answers is not in the document]
Comment (mghiremath) [reply data/tasks.js]: refactored everywhere

File: data/tasks.js (hunk 2 above)
Comment (mghiremath) [reply data/tasks.js]: done

File: public/scripts/week-calendar.js (hunk 8 above)
Comment (mghiremath) [reply public/scripts/week-calendar.js]: updated the routes to take user id, returning user ID and task id

**Review by sanjeet-jain — APPROVED** [review]

**Pull request merged**